azerothcore/azerothcore-wotlk
# fix(Scripts/Spell): combustion will now properly go on cooldown when the buff is manually cancelled

## Changes Proposed:
-  Ported the changes from TrinityCore/TrinityCore@e07a322
-  

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/7182
- Closes https://github.com/chromiecraft/chromiecraft/issues/1312

## SOURCE:


## Tests Performed:

- Added some levels to the character
- Learn combustion on the fire talent tree
- Use it, cast 1 or 2 spells and cancel de aura with right click
- Stacks should stop and the cooldown should start

Video proof

https://streamable.com/l504rd

## How to Test the Changes:


1. Added some levels to the character
2. Learn combustion on the fire talent tree
3. Use it, cast 1 or 2 spells and cancel de aura with right click
4. Stacks should stop and the cooldown should start


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
